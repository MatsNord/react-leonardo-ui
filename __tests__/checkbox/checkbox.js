import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../../src/checkbox';

test('Normal checkbox renders correctly', () => {
  const tree = renderer.create(
    <Checkbox checked />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox with states renders correctly', () => {
  const tree = renderer.create(
    <Checkbox checked active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Checkbox checked variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox passes props correctly', () => {
  const onChange = jest.fn(() => {});

  const tree = renderer.create(
    <Checkbox checked className="my-class" data-x="x" title="Title" onChange={onChange} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  const input = tree.children[0];
  input.props.onChange();
  expect(onChange.mock.calls.length).toBe(1);
});
