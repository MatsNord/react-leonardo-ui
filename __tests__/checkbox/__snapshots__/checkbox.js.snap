exports[`test Checkbox passes props correctly 1`] = `
<label
  className="lui-checkbox my-class"
  htmlFor={undefined}
  title="Title">
  <input
    checked={true}
    className="lui-checkbox__input"
    data-x="x"
    onChange={[Function]}
    type="checkbox" />
  <div
    className="lui-checkbox__check-wrap">
    <span
      className="lui-checkbox__check" />
    <span
      className="lui-checkbox__check-text" />
  </div>
</label>
`;

exports[`test Checkbox with modifiers renders correctly 1`] = `
<label
  className="lui-checkbox lui-checkbox--inverse"
  htmlFor={undefined}
  title={undefined}>
  <input
    checked={true}
    className="lui-checkbox__input"
    type="checkbox" />
  <div
    className="lui-checkbox__check-wrap">
    <span
      className="lui-checkbox__check" />
    <span
      className="lui-checkbox__check-text" />
  </div>
</label>
`;

exports[`test Checkbox with states renders correctly 1`] = `
<label
  className="lui-checkbox lui-active"
  htmlFor={undefined}
  title={undefined}>
  <input
    checked={true}
    className="lui-checkbox__input"
    disabled={true}
    type="checkbox" />
  <div
    className="lui-checkbox__check-wrap">
    <span
      className="lui-checkbox__check" />
    <span
      className="lui-checkbox__check-text" />
  </div>
</label>
`;

exports[`test Normal checkbox renders correctly 1`] = `
<label
  className="lui-checkbox"
  htmlFor={undefined}
  title={undefined}>
  <input
    checked={true}
    className="lui-checkbox__input"
    type="checkbox" />
  <div
    className="lui-checkbox__check-wrap">
    <span
      className="lui-checkbox__check" />
    <span
      className="lui-checkbox__check-text" />
  </div>
</label>
`;
